source/Agents/PathAgent.cpp

```diff
@@ -0,0 +1,231 @@
+  }
+}
+
+/**
+ * Moves the agent by applying the current offset
+ * @return whether the update succeeded
+ */
+size_t PathAgent::SelectAction(cse491::WorldGrid const& /* grid*/,
+                               cse491::type_options_t const& /* type_options*/,
+                               cse491::item_set_t const& /* item_set*/,
+                               cse491::agent_set_t const& /* agent_set*/) {
+  Update();
+  return action_map["move_self"];
+}
+
+/**
+ * Assigns the offsets_member to a new series of offsets
+ * @param offsets collection of grid positions used as the new offsets
+ * @param start_index which offset to start indexing into (beginning by default)
+ * @return self
+ * @attention throws an `std::invalid_argument` when an invalid start index is provided
+ */
+PathAgent& PathAgent::SetPath(std::vector<cse491::GridPosition> && offsets, size_t start_index) {
+  offsets_ = offsets;
+  index_ = start_index;
+  if (static_cast<size_t>(index_) >= offsets_.size()) {
+    std::ostringstream what;
+    what << "Out of bounds offset index to begin from: " << index_ << ", number of offsets: " << offsets_.size();
+    throw std::invalid_argument(what.str());
+  }
+  return *this;
+}
+
+/**
+ * Assigns the offsets_ member to a new series of offsets, taking a command string
+ * @param commands formatted string of commands used as offsets
+ * @param start_index which command to begin indexing into (first command by default)
+ * @return self
+ * @attention throws an `std::invalid_argument` when mis-formatted commands an invalid index is provided
+ */
+PathAgent& PathAgent::SetPath(std::string_view commands, size_t start_index) {
+  offsets_.clear();
+  return SetPath(str_to_offsets(commands), start_index);
+}
+
+/**
+ * Returns an immutable reference to this agent's current path
+ * @return sequence of offsets
+ */
+std::vector<cse491::GridPosition> const& PathAgent::GetPath() const {
+  return offsets_;
+}
+
+/**
+ * Converts a string to a sequence of offsets
+ *
+ * This convenience method takes a string with a special formatting that allows one to specify
+ * a sequence of whitespace-separated inputs in linear directions.
+ * The format is [steps[*]]<direction> where
+ * `steps` is a positive integer and optional (assumed to be 1 by default)
+ * star `*` represents scaling the movement by `steps`. Optional, but cannot be used if `steps` is not provided
+ * if the star is not present, then `steps` individual offsets are created in the direction `direction`
+ * Example: "n w 3e 10*s 5*w x" should create the sequence of offsets
+ * {-1, 0}, {0, 1}, {0, -1}, {0, -1}, {0, -1}, {10, 0}, {0, 5}, {0, 0}
+ * @param commands string in a format of sequential directions
+ * @note throws an `std::invalid_argument` when input string is poorly formatted
+ * @note this includes when a negative integer is passed as `steps`. If a zero is used, treated as the default (one)
+ */
+std::vector<cse491::GridPosition> str_to_offsets(std::string_view commands) {
```

Comment:
Should be renamed to StrToOffsets to match other function naming conventions

Reply:
Function has been renamed